### PR TITLE
Add a new recipe

### PR DESCRIPTION
**Pull request by tmeschter**
Add a new recipe describing how to transform or filter existing dataflows.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/9097)

### CODE REVIEW

File: docs/repo/joy-of-dataflow.md [hunk 1]
```diff
@@ -34,4 +38,29 @@ Do not use this recipe if the output data can be produced by a 1-to-1 transforma
...
+3. Override and implement the `LinkExternalInput` method. This is where the dataflow block(s) will be created and where you will perform any setup required to start producing data.
+    1. Use `DataflowBlockSlim.CreateTransformBlock` to create a transform block that takes in `IProjectVersionedValue<U>` (where `U` is the input data) and produces `IProjectVersionedValue<T>` (again, `T` is the type of data you want to expose).
+    2. Since the intent here is to transfrom data from existing sources the version keys and values for your output data will generally be the same as the input. To avoid boilerplate code around `IProjectVersionedValue<>` handling, pass in the lambda `update => update.Derive(Transform)` for the `transformFunction` parameter of `CreateTransformBlock`. The `Derive` extension method handles calling the `Transform` method (to be defined in a later step) to convert `U`s to `T`s and producing the final `IProjectVersionedValue<>` with the version keys and values copied over.
+    3. Link your input dataflow to the transform block, and the transform block to the `targetBlock` passed in to `LinkExternalInput`. Store the resulting link objects in a `DisposableBag`.
```
Comment (haileymck) [inline docs/repo/joy-of-dataflow.md:54]: what do you mean by link?

**Review by haileymck — APPROVED** [review]

File: docs/repo/joy-of-dataflow.md (hunk 1 above)
Comment (smitpatel) [reply docs/repo/joy-of-dataflow.md:54]: Connect 2 blocks.

Comment (drewnoakes) [reply docs/repo/joy-of-dataflow.md:54]: Via the `LinkTo` method.

File: docs/repo/joy-of-dataflow.md [hunk 2]
```diff
@@ -2,13 +2,17 @@
 
 Recipes for doing all sorts of things with System.Threading.Dataflow in the Project System.
 
+## Table of Contents
+
+- [How to create a new data source](#how-to-create-a-new-data-source)
+- [How to filter or transform an existing data source](#how-to-filter-or-transform-an-existing-data-source)
```
[the remark this answers is not in the document]
Comment (tmeschter) [reply docs/repo/joy-of-dataflow.md:8]: I'm not aiming for fully general documentation about dataflow or dataflow in CPS. Rather, as I'm encountering specific scenarios where I'm using dataflow (especially ones I've encountered multiple times) I'm trying to write down the important parts in order to provide some guidance. There's certainly a need for background material and general information, but I think that would be a separate document, or a separate part of a much longer document.

That being said I'm open to moving this to VSProjectSystem if/when we can make it part of a larger documentation push.

File: docs/repo/joy-of-dataflow.md [hunk 3]
```diff
@@ -2,13 +2,17 @@
...
+
+
 ## How to create a new data source
 
 Use this recipe when you want to expose a fundamentally new _type_ of data that isn't available from some existing dataflow.
 
-Do not use this recipe if the output data can be produced by a 1-to-1 transformation of an existing dataflow, or by subsetting or filtering an existing dataflow.
-
-> TODO: Link to the appropriate recipes for transforming and filtering dataflows.
+Do not use this recipe if the output data can be produced by a 1-to-1 transformation of an existing dataflow, or by subsetting or filtering an existing dataflow. See [How to filter or transform an existing data source](#how-to-filter-or-transform-an-existing-data-source) instead.
```
[the remark this answers is not in the document]
Comment (tmeschter) [reply docs/repo/joy-of-dataflow.md:15]: 👍🏻

**Review by drewnoakes — APPROVED** [review]

**Pull request merged**